API/Configuration/GlobalExceptionHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System.Net;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Configuration;
+
+public class GlobalExceptionHandler : IExceptionHandler
+{
+    private readonly ILogger<GlobalExceptionHandler> _logger;
+    public GlobalExceptionHandler(ILogger<GlobalExceptionHandler> logger)
+    {
+        _logger = logger;
+    }
+    public async ValueTask<bool> TryHandleAsync(HttpContext httpContext, Exception exception, CancellationToken cancellationToken)
+    {
+        var internalServerError = HttpStatusCode.InternalServerError;
+        var path = httpContext.Request.Path;
+        ProblemDetails errorResponse = new ProblemDetails();
+        _logger.LogError("Exception happend at {path} for connection id: {id}, exception trace: {type}  Message: {exception}.", path, httpContext.Connection.Id, exception.StackTrace, exception.Message);
+        switch (exception)
+        {
+            default:
+                errorResponse.Type = "https://datatracker.ietf.org/doc/html/rfc9110#section-15.6.1";
+                errorResponse.Title = nameof(HttpStatusCode.InternalServerError);
+                errorResponse.Status = (int)internalServerError;
+                errorResponse.Detail = exception.Message;
```

Comment:
As mentioned in the PR, I could see the point in not exposing this to the frontend.

If the GlobalExceptionHandler is meant to catch faults we are unaware of, it has no relevance for users of the system IMO.